Frameworks/Foundation/NSData.mm

```diff
@@ -609,27 +614,14 @@ - (NSString*)description {
     // length * 2   = 2 chars (hex representation) for each byte
     // length / 4   = spaces - one every 4 bytes
     // 16           = <> and misc
-    // if the length > 1024, describe only the first 512 bytes and the last 512 bytes,
-    // and add a "... " (4 bytes) in the middle
-    const NSUInteger lengthToUse = std::min(1024u, length);
-    const int tmpBufSize = 16 + lengthToUse * 2 + (lengthToUse / 4) + (length > 1024 ? 4 : 0);
+    const int tmpBufSize = 16 + length * 2 + (length / 4);
 
     std::vector<char> tmpBuf(tmpBufSize);
     int tmpBufLen = 0;
 
     tmpBuf[tmpBufLen++] = '<';
 
     for (auto i = 0; i < length;) {
-        // skip the middle portion, if length > 1024, and go directly to the last 512 bytes
```

Comment:
IIRC this was needed / confirmed to be used. Why did you remove it?

---
In reply to: [73397672](https://github.com/Microsoft/WinObjC/pull/798#discussion_r73397672) [](ancestors = 73397672)